bbid.py

```diff
@@ -94,7 +100,8 @@ def backup_history(*args):
 	download_history=open(output_dir + '/download_history.pickle','wb')
 	pickle.dump(tried_urls,download_history)
 	pickle.dump(finished_keywords, download_history)
-	pickle.dump(image_md5s, download_history)
+	copied_image_md5s = dict(image_md5s)  # To resolve a concurrency issue
+	pickle.dump(copied_image_md5s, download_history)
```

Comment:
Again, is this the actual problem you experienced? I don't see the issue with updating the variable while pickle is doing its stuff